src/passes/OptimizeInstructions.cpp

```diff
@@ -71,7 +72,16 @@ struct PatternDatabase {
   };
 };
 
-static PatternDatabase database;
+static PatternDatabase* database = nullptr;
+
+static void ensureDatabase() {
+  if (!database) {
+    // we must only ever create one database
+    static OnlyOnce onlyOnce;
+    onlyOnce.verify();
+    database = new PatternDatabase;
```

Comment:
Yeah, maybe you're right. But since this `database` is I am guessing hot, we should measure first.